pkg/cloud/services/provider/provider.go

```diff
@@ -169,34 +182,3 @@ type project struct {
 type projects struct {
 	Projects []project `json:"projects"`
 }
-
-func GetProjectID(client *gophercloud.ProviderClient, clientOpts *clientconfig.ClientOpts) (string, error) {
```

Comment:
looks like we are going to drop the project (for normal case)? 

I understand we want to support application cred, but if this logic is removed, is it replaced by line 115-125 above?
but that code comment said it's for application cred only...

Reply:
Yes, we should never have to pull project id from clientOpts. It should always be returned in the authentication response.

Reply:
@mdbooth is right, in a previous version of the patch I only added the new flow for application credentials, and this is where the comment is coming from.

Now that it's used for all auth mechanisms, the comment is outdated and I will remove it in the next push.